### PR TITLE
Adds Inf and NaN as Numbers to Histogram in Promql Testing Framework

### PR DESCRIPTION
**Pull request by NeerajGartia21**
Currently Promql Testing Framework doesn't support `Inf` and `NaN` as numbers in histogram description such as `sum` and `count`. Descriptions like- `sum: NaN` and `count: Inf` results in error and this PR fixes this issue.

### CODE REVIEW

File: promql/parser/lex.go [hunk 1]
```diff
@@ -568,10 +568,16 @@ Loop:
 					return lexHistogram
 				}
 				l.errorf("missing `:` for histogram descriptor")
-			} else {
-				l.errorf("bad histogram descriptor found: %q", word)
+				break Loop
 			}
-
+			// current word is Inf or NaN
```
Comment (beorn7) [inline promql/parser/lex.go:573]: Nit: Use normal capitalization and punctuation.

```suggestion
			// Current word is Inf or NaN.
```

File: promql/parser/lex_test.go [hunk 2]
```diff
@@ -561,6 +561,35 @@ var tests = []struct {
 				},
 				seriesDesc: true,
 			},
+			{ // series with sum as -Inf and count as NaN
```
Comment (beorn7) [inline promql/parser/lex_test.go:564]:
```suggestion
			{ // Series with sum as -Inf and count as NaN.
```

**Review by beorn7 — APPROVED** [review]
Thank you very much. 

I'm not an expert when it comes to the lexer, but I assume this is the way to go.

Note that generally, we try to use normal English syntax in comments. I will commit my nits directly, so no need for you to update this PR.

**Comment by NeerajGartia21** [comment]
Thanks @beorn7 for merging it and I'll follow general english syntax from now on.

**Pull request merged**